### DISCUSSION TITLE
problem when runing SOLWEIG

### DISCUSSION THREAD

**Discussion by yaogan2024** [UMEP-dev/UMEP#673] [General]
problem when runing SOLWEIG:
QGIS版本：3.34.9-Prizren
QGIS代码版本：cbbc7bdf
Qt版本：5.15.13
Python版本：3.12.4
GDAL版本：3.9.1
GEOS版本：3.12.2-CAPI-1.18.2
PROJ版本：Rel. 9.4.0, March 1st, 2024
PDAL版本：2.6.3 (git-version: b5523a)
算法启动于: 2024-11-20T20:00:56
正在启动 'Outdoor Thermal Comfort: SOLWEIG v2022a' 算法…
输入参数：
{ 'ABS_L' : 0.95, 'ABS_S' : 0.7, 'ACTIVITY' : 80, 'AGE' : 35, 'ALBEDO_GROUND' : 0.15, 'ALBEDO_WALLS' : 0.2, 'CLO' : 0.9, 'CONIFER_TREES' : False, 'CYL' : True, 'EMIS_GROUND' : 0.95, 'EMIS_WALLS' : 0.9, 'HEIGHT' : 180, 'INPUTMET' : 'F:\\Tmrt\\test_0706.txt', 'INPUT_ANISO' : '', 'INPUT_ASPECT' : 'F:/Tmrt/test/test_wallaspect.tif', 'INPUT_CDSM' : 'F:/Tmrt/test/canopyheight.13331.tif', 'INPUT_DEM' : 'F:/Tmrt/test/DEM.13331.tif', 'INPUT_DSM' : 'F:/Tmrt/test/builtup.13331.tif', 'INPUT_HEIGHT' : 'F:/Tmrt/test/test_wallheight.tif', 'INPUT_LC' : None, 'INPUT_SVF' : 'F:\\Tmrt\\test_svf\\svfs.zip', 'INPUT_TDSM' : None, 'INPUT_THEIGHT' : 25, 'LEAF_END' : 300, 'LEAF_START' : 97, 'ONLYGLOBAL' : True, 'OUTPUT_DIR' : 'F:\\Tmrt\\1115', 'OUTPUT_KDOWN' : False, 'OUTPUT_KUP' : False, 'OUTPUT_LDOWN' : False, 'OUTPUT_LUP' : False, 'OUTPUT_SH' : False, 'OUTPUT_TMRT' : True, 'OUTPUT_TREEPLANTER' : False, 'POI_FIELD' : '', 'POI_FILE' : None, 'POSTURE' : 0, 'SAVE_BUILD' : False, 'SENSOR_HEIGHT' : 10, 'SEX' : 0, 'TRANS_VEG' : None, 'USE_LC_BUILD' : False, 'UTC' : 0, 'WEIGHT' : 75 }

Longitude derived from DSM: 4385662.9656
Latitude derived from DSM: 1526310.5532
Vegetation scheme activated
Digital Evevation Model (DEM) included negative values. DEM raised with 18.406391143798828m.
Sky View Factor rasters loaded
Meteorological data successfully loaded
Calculating sun positions for each time step
Isotropic sky
Writing settings for this model run to specified output folder (Filename: RunInfoSOLWEIG_YYYY_DOY_HHMM.txt)
Executing main model
Traceback (most recent call last):
File "C:\Users/DA/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\processing_umep\processor\solweig_algorithm.py", line 1177, in processAlgorithm
saveraster(gdal_dsm, outputDir + '/Tmrt_' + str(int(YYYY[0, i])) + '_' + str(int(DOY[i]))
File "C:\Users/DA/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\processing_umep\util\misc.py", line 34, in saveraster
outBand = outDs.GetRasterBand(1)
^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'GetRasterBand'

Execution failed after 855.10 秒 (14 分钟 15 秒)

**Comment by biglimp** [comment]
Your coordinates (lat, lon) is in correct, most probably because you are using a coordinate system in QGIS which cannot be translated to accurate lat, lon. Try changing to a more common CRS, e.g. UTM.

**Reply by yaogan2024** [reply]
I change to UTM, then a new error occurs:
Traceback (most recent call last):
File "C:\Users/DA/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\processing_umep\processor\solweig_algorithm.py", line 1006, in processAlgorithm
dRad, Kside = so.Solweig_2022a_calc(
^^^^^^^^^^^^^^^^^^^^^^
File "C:\Users/DA/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\processing_umep\functions\SOLWEIGpython\Solweig_2022a_calc_forprocessing.py", line 196, in Solweig_2022a_calc
gvfalbnoshW, gvfLupN, gvfalbN, gvfalbnoshN, gvfSum, gvfNorm = gvf_2018a(wallsun, walls, buildings, scale, shadow, first,
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Users/DA/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\processing_umep\functions\SOLWEIGpython\gvf_2018a.py", line 32, in gvf_2018a
_, gvfLupi, gvfalbi, gvfalbnoshi, gvf2 = sunonsurface_2018a(azimuthA[j], scale, buildings, shadow, sunwall,
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Users/DA/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\processing_umep\functions\SOLWEIGpython\sunonsurface_2018a.py", line 165, in sunonsurface_2018a
gvf1 = ((weightsumwall_first + weightsumsh_first) / (first + 1)) * wallsuninfluence_first + \
^^^^^^^^^^^^^^^^^^^
UnboundLocalError: cannot access local variable 'weightsumwall_first' where it is not associated with a value

Execution failed after 51.78 秒

**Reply by biglimp** [reply]
There should be a RunInfo file in your output folder. What does it say?

**Reply by yaogan2024** [reply]
This file provides run settings for the SOLWEIG run initiated at: Fri, 22 Nov 2024 10:38:37
Version: SOLWEIG v2022a

SURFACE DATA
Digital surface model (DSM): F:/Tmrt/11.11/11.11/builtup.13331.tif
Model domain: rows = 302, columns = 360
Projected reference system: WGS 84 / UTM zone 48N
Geographical coordinate system: WGS 84
Latitude: 39.01453232703777
Longitude: 116.83747317452357
UTC: 0.0
Transmissivity of light through vegetation: 0.03
Digital vegetation canopy model (CDSM): F:/Tmrt/11.11/11.11/canopyheight.13331.tif
Trunkzone estimated from CDSM
Trunkzone as percent of canopy height: 0.25
Landcover scheme inactive

DEM used to identify buildings

METEOROLOGICAL FORCING DATA
Meteorological file: F:\Tmrt\test_0706.txt
Diffuse and direct shortwave radiation estimated from global radiation

HUMAN EXPOSURE PARAMETERS
Absorption, shortwave radiation: 0.7
Absorption, longwave radiation: 0.95
Posture of human body: Standing
ENVIRONMENTAL PARAMETERS
Albedo of walls: 0.2
Albedo of ground (not used if land cover scheme is active): 0.15
Emissivity (walls): 0.9
Emissivity of ground (not used if land cover scheme is active): 0.95

ADDITIONAL SETTINGS
Human considered as a standing cylinder
Isotropic sky

**Reply by biglimp** [reply]
Your UTC looks wrong. That should be set according to local time and location. In your case, maybe 8?
